src/energiapy/model/constraints/failure.py

```diff
@@ -16,18 +16,23 @@
 from ...utils.scale_utils import scale_list
 
 
-def constraint_nameplate_production_failure(instance: ConcreteModel, fail_factor: dict = {}, network_scale_level: int = 0, scheduling_scale_level: int = 0) -> Constraint:
+def constraint_nameplate_production_failure(instance: ConcreteModel, fail_factor: dict = None, network_scale_level: int = 0, scheduling_scale_level: int = 0) -> Constraint:
```

Comment:
changed default mutable argument to None, then a non  guard at the beginning of the function